app/src/main/java/com/pileproject/drive/execution/ExecutionCondition.java

```diff
@@ -17,28 +17,19 @@
 package com.pileproject.drive.execution;
 
 
-import android.content.Context;
-
-import com.pileproject.drive.R;
-import com.pileproject.drive.app.DriveApplication;
 import com.pileproject.drive.programming.visual.block.BlockBase;
 
 import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.Map;
 import java.util.Stack;
 
 /**
- * This class has the condition of execution
- *
- * @author <a href="mailto:tatsuyaw0c@gmail.com">Tatsuya Iwanari</a>
- * @version 1.0 7-July-2013
+ * A container class that has the condition of program execution.
  */
 public class ExecutionCondition {
```

Comment:
I'm sorry for late replay 😢 

Originally, I prepared `pushSelectionResult(...)` as a helper method. With it, users (mainly the method is used in *Block classes) can use it without considering the program count. But you're right and I will prepare getters and make ifStack private.